src/reconcile/rec_visibility.c

```diff
@@ -419,6 +418,21 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
 
             /* Find the update this tombstone applies to. */
             if (!__wt_txn_upd_visible_all(session, upd)) {
+                /*
+                 * Handle double tombstone scenario that can happen in the history store when we add
+                 * zero timestamped tombstones to the front of the update chain.
+                 */
+                if (upd->next != NULL && upd->next->type == WT_UPDATE_TOMBSTONE) {
```

Comment:
I know in the history store we don't have aborted transactions but I think we should better also handle the case that T@0 -> aborted -> T@10 -> U@20 here just in case.

Reply:
Good point.